File: TI1.2 SOAP_DOTNET_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Dotnet_GR06/ConUni_Soap_Dotnet_GR06/Servicios/ConversorTemperatura.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ConUni_Soap_Dotnet_GR06.Servicios
{
    /// <summary>
    /// Servicio de conversión de temperaturas
    /// </summary>
    public class ConversorTemperatura
    {
        public double CelsiusAFahrenheit(double celsius)
        {
            return (celsius * 9.0 / 5.0) + 32;
        }

        public double FahrenheitACelsius(double fahrenheit)
        {
            return (fahrenheit - 32) * 5.0 / 9.0;
        }

        public double CelsiusAKelvin(double celsius)
        {
            return celsius + 273.15;
        }

        public double KelvinACelsius(double kelvin)
        {
            return kelvin - 273.15;
        }

        public double FahrenheitAKelvin(double fahrenheit)
        {
            return CelsiusAKelvin(FahrenheitACelsius(fahrenheit));
        }

        public double KelvinAFahrenheit(double kelvin)
        {
            return CelsiusAFahrenheit(KelvinACelsius(kelvin));
        }
    }
}

File: Deberes_PP_Arquitectura-entregable/TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/services/ConversorTemperatura.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.services;

/**
 *
 * @author josue
 */
public class ConversorTemperatura {
    public double celsiusAFahrenheit(double celsius) {
        return (celsius * 9/5) + 32;
    }
    
    public double fahrenheitACelsius(double fahrenheit) {
        return (fahrenheit - 32) * 5/9;
    }
    
    public double celsiusAKelvin(double celsius) {
        return celsius + 273.15;
    }
    
    public double kelvinACelsius(double kelvin) {
        return kelvin - 273.15;
    }
    
    public double fahrenheitAKelvin(double fahrenheit) {
        return celsiusAKelvin(fahrenheitACelsius(fahrenheit));
    }
    
    public double kelvinAFahrenheit(double kelvin) {
        return celsiusAFahrenheit(kelvinACelsius(kelvin));
    }    
}
